indigo_app/views/works.py

```diff
@@ -71,47 +71,11 @@ def get_context_data(self, **kwargs):
         return context
 
     def get_work_timeline(self, work):
-        # get initial, amendment, consolidation dates
-        dates = work.possible_expression_dates()
-        # add assent, publication, repeal, commencement dates
-        other_dates = [
-            ('assent_date', work.assent_date),
-            ('publication_date', work.publication_date),
-            ('repealed_date', work.repealed_date)
-        ]
-        for c in work.commencements.all():
-            other_dates.append(('commencement_date', c.date))
-
-        for name, date in other_dates:
-            if date:
-                if date not in [entry['date'] for entry in dates]:
-                    dates.append({
-                        'date': date,
-                        name: True,
-                    })
-                else:
-                    # add to existing events (e.g. if publication and commencement dates are the same)
-                    for entry in dates:
-                        if entry['date'] == date:
-                            entry[name] = True
-
-        dates.sort(key=lambda x: x['date'], reverse=True)
-
-        # add expressions and commencement and amendment objects
-        for event in dates:
-            date = event['date']
-            if event.get('commencement_date'):
-                event['commencements'] = work.commencements.filter(date=date).all()
-            if event.get('amendment'):
-                event_amendments = work.amendments.filter(date=date)
-                if len(event_amendments) > 1:
-                    event_amendments = Amendment.order_further(event_amendments)
-                event['amendments'] = event_amendments
-            if event.get('consolidation'):
-                event['consolidations'] = work.arbitrary_expression_dates.filter(date=date).all()
-            event['expressions'] = work.expressions().filter(expression_date=date).all()
-
-        return dates
+        timeline = work.get_timeline()
+        # add expressions
+        for entry in timeline:
+            entry.expressions = work.expressions().filter(expression_date=entry.date).all()
```

Comment:
This is actually pretty inefficient. Better to load all the expressions once, index by date, and then filter them here.

```python
expressions = list(work.expressions().all())
...
entry.expressions = [e for in in expressions if e.expression_date == e.date]
```